### PR TITLE
Migrate the toml checker to Redocly (https://xrpl.org/xrp-ledger-toml-checker.html)

### PR DESCRIPTION
**Pull request by JST5000**
This migrates this page: https://xrpl.org/xrp-ledger-toml-checker.html

It primarily involves migrating:
1. `xrpl-toml-checker.page.tsx`
2. `xrp-ledger-toml-checker.js`
3. `iarna-toml-parse.js`
4. (And moving over the test file `xrp-ledger-toml-checker-test.js`)

### CODE REVIEW

File: content/dev-tools/toml-checker/ListTomlFields.tsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+import { Client } from 'xrpl'
+
+// Decode a hexadecimal string into a regular string, assuming 8-bit characters.
+// Not proper unicode decoding, but it'll work for domains which are supposed
+
+import React = require("react");
+import { CLASS_GOOD } from "./LogEntry";
+import { AccountFields } from "./XrplToml";
+
+// to be a subset of ASCII anyway.
```
Comment (mDuo13) [inline content/dev-tools/toml-checker/ListTomlFields.tsx]: This comment is the continuation of the ones on lines 4-5. It probably shouldn't have a bunch of imports in between.

File: content/dev-tools/xrpl-toml-checker.page.tsx
Comment (mDuo13) [inline content/dev-tools/xrpl-toml-checker.page.tsx:1]: Rename to `xrp-ledger-toml-checker.page.tsx` for consistency with the old URL and the name of the file it's checking, and move under `resources/dev-tools/`

**Review by mDuo13 — CHANGES_REQUESTED** [review]
1. To match the location of the other tools, please put it under `resources/dev-tools/`.
2. Modify the `content/sidebars.yaml` file to add the `page: path/to/file.page.tsx` line. You can skip the `href` line which doesn't actually do anything in my experience.
3. I think there's some kind of error handling problem going on. Testing it on `mduo13.com` on leads to a blank result (with errors in the console). The tool should display the results _especially_ if parsing the TOML or looking up the accounts didn't go well, since the point of this tool is to detect and point out problems.

**Comment by JST5000** [comment]
> * To match the location of the other tools, please put it under `resources/dev-tools/`.
> * Modify the `content/sidebars.yaml` file to add the `page: path/to/file.page.tsx` line. You can skip the `href` line which doesn't actually do anything in my experience.
> * I think there's some kind of error handling problem going on. Testing it on `mduo13.com` on leads to a blank result (with errors in the console). The tool should display the results _especially_ if parsing the TOML or looking up the accounts didn't go well, since the point of this tool is to detect and point out problems.

1st two are done, 3rd one I could not reproduce unfortunately.

**Comment by mDuo13** [comment]
Approved and merged with the same sidebar-related changes as #2270.

**Pull request merged**